### PR TITLE
fix(core): only stop daemon if enabled

### LINKED ISSUE #27971: nx reset hangs on CI
### Current Behavior

Context: angular monorepo, latest nx version, process hangs indefinitely in pipeline (jenkins)

```
[2024-09-18T12:04:52.267Z] info  npmExecuteScripts - > nx reset
[2024-09-18T12:04:52.267Z] info  npmExecuteScripts - 
[2024-09-18T12:04:52.267Z] info  npmExecuteScripts - 
[2024-09-18T12:04:52.267Z] info  npmExecuteScripts -  NX   Resetting the Nx cache and stopping the daemon.
[2024-09-18T12:04:52.267Z] info  npmExecuteScripts - 
[2024-09-18T12:04:52.267Z] info  npmExecuteScripts - 
[2024-09-18T12:04:52.267Z] info  npmExecuteScripts - (node:851) ExperimentalWarning: WASI is an experimental feature and might change at any time
[2024-09-18T12:04:52.267Z] info  npmExecuteScripts - (Use `node --trace-warnings ...` to show where the warning was created)
[2024-09-18T12:04:52.267Z] info  npmExecuteScripts - 
[2024-09-18T12:04:52.267Z] info  npmExecuteScripts -  NX   Watcher is not a constructor
[2024-09-18T12:04:52.267Z] info  npmExecuteScripts - 
[2024-09-18T12:04:52.267Z] info  npmExecuteScripts - 
[2024-09-18T12:04:52.267Z] info  npmExecuteScripts -  NX   Successfully reset the Nx workspace.
[2024-09-18T12:04:52.267Z] info  npmExecuteScripts -
```

after this, it never proceeds to the next step

### Expected Behavior

`nx reset` should not get stuck indefinitely.

### GitHub Repo

_No response_

### Steps to Reproduce

1. run `nx reset` on the pipeline

I'm aware this isn't really helpful, sorry

### Nx Report

```shell
(note that this is on my local machine, not the CI)

Node           : 20.14.0
OS             : win32-x64
Native Target  : x86_64-windows
npm            : 10.2.3

nx                 : 19.7.2
@nx/js             : 19.7.2
@nx/jest           : 19.7.2
@nx/linter         : 19.7.2
@nx/eslint         : 19.7.2
@nx/workspace      : 19.7.2
@nx/angular        : 19.7.2
@nx/cypress        : 19.7.2
@nx/devkit         : 19.7.2
@nx/eslint-plugin  : 19.7.2
@nrwl/tao          : 19.7.2
@nx/web            : 19.7.2
@nx/webpack        : 19.7.2
typescript         : 5.5.4
---------------------------------------
Registered Plugins:
@nx/eslint/plugin
@nx/jest/plugin
@nx/cypress/plugin
---------------------------------------
Community plugins:
@fundamental-ngx/cdk      : 0.52.1
@fundamental-ngx/core     : 0.52.1
@fundamental-ngx/platform : 0.52.1
```

### Failure Logs

_No response_

### Package Manager Version

_No response_

### Operating System

- [ ] macOS
- [X] Linux
- [ ] Windows
- [ ] Other (Please specify)

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by xiongemi**
## Related Issue(s)

Fixes https://github.com/nrwl/nx/issues/27971

### CODE REVIEW

File: packages/nx/src/command-line/reset/reset.ts [hunk 1]
```diff
@@ -66,7 +66,11 @@ export async function resetHandler(args: ResetCommandOptions) {
     }
   }
   if (all || args.onlyCloud) {
-    await resetCloudClient();
+    try {
+      await resetCloudClient();
+    } catch {
+      errors.push('Failed to reset the Nx Cloud client.');
```
Comment (FrozenPandaz) [inline packages/nx/src/command-line/reset/reset.ts:72]: Include the error that got thrown.

File: packages/nx/src/command-line/reset/reset.ts [hunk 2]
```diff
@@ -1,5 +1,5 @@
 import { rmSync } from 'node:fs';
-import { daemonClient } from '../../daemon/client/client';
```
Comment (FrozenPandaz) [inline packages/nx/src/command-line/reset/reset.ts:2]: This should be fine

File: packages/nx/src/command-line/reset/reset.ts [hunk 3]
```diff
@@ -81,7 +85,11 @@ export async function resetHandler(args: ResetCommandOptions) {
   }
 }
 
-function killDaemon() {
+async function killDaemon(): Promise<void> {
+  const { daemonClient } = await import('../../daemon/client/client');
+  if (!daemonClient.enabled) {
+    return;
+  }
   return daemonClient.stop();
```
Comment (FrozenPandaz) [inline packages/nx/src/command-line/reset/reset.ts:93]:
```suggestion
  const { daemonClient } = await import('../../daemon/client/client');
  if (daemonClient.enabled()) {
    return daemonClient.stop();
  }
```

**Review by FrozenPandaz — CHANGES_REQUESTED** [review]

**Review by FrozenPandaz — APPROVED** [review]

**Comment by dunika** [comment]
Thanks!

**Pull request merged**